tests/store/test_version_filters.py

```diff
@@ -0,0 +1,57 @@
+from housekeeper.store.api.core import Store
+from housekeeper.store.models import Version
+from housekeeper.store.version_filters import filter_version_by_id
+
+
+def test_filter_version_by_id_returns_the_correct_version(populated_store: Store):
+    """Test getting version by version ID."""
+
+    # GIVEN a store with a version
+    version: Version = populated_store.session.query(Version).first()
+    assert version
+
+    version_id: int = version.id
+
+    # WHEN retrieving the version by ID
+    filtered_version: Version = filter_version_by_id(
+        versions=populated_store.session.query(Version),
+        version_id=version_id,
+    ).first()
+
+    # THEN a version should be returned
+    assert isinstance(filtered_version, Version)
+
+    # THEN the id should match
+    assert filtered_version.id == version_id
+
+
+def test_filter_version_by_id_with_invalid_id_returns_empty_query(populated_store: Store):
+    """Test that an empty result is returned when using an invalid version ID."""
+
+    # GIVEN a store with some versions
+    versions_query = populated_store.session.query(Version)
+    assert versions_query.count() > 0
+
+    # GIVEN an invalid id that doesn't exist in the store
+    invalid_version_id = versions_query.order_by(Version.id.desc()).first().id + 1
```

Comment:
Should`invalid_version_id` be a fixture for reuse and maybe it is ok to hard code the return from the `order_by` in the fixture? An invalid version id should not change much whatever we do to the rest of the code

Reply:
Hm, invalid was a bad name. I'll keep it as it is and change the name to non_existing_id for now.

Reply:
I'm not sure if it should be a fixture, it would only be guaranteed to not exist for the specific store the fixture uses.